### ISSUE TITLE
Unable to "redock" Windows on Ubuntu 22.04 + Wayland

### ISSUE THREAD

**Issue by nyoungbq** [githubuser0xFFFF/Qt-Advanced-Docking-System#483]
# Bug Report
OS: Ubuntu 22.04.1 LTS
Desktop Client: Gnome v42.5
Windowing System: Wayland
Compiler: GCC v11.3.0
Build System: CMake + Ninja

I ran into an issue on Linux, where I can undock a widget from the main window and extract it out into its own window, but once its out it is unable to redock with the main window. The selection menu just doesn't even appear to snap it back into place. I tested this out on your example program - "AdvancedDockingSystemDemo" and the issue is also present. I am getting around this in my application by resetting my window preferences .json to defaults any time I need to redock, but this is unsustainable long term.

## Steps to Reproduce
1. Undock window from primary
2. Drop window outside primary window
3. Attempt to drag external Window back into primary 

## Here is an example:
[Screencast from 02-09-2023 02:14:46 PM.webm](https://user-images.githubusercontent.com/109472155/217914812-d711627d-851e-4ba8-b1ba-e96a1e78a031.webm)

## Other Useful Information
This seems to be platform dependent, our other tested platforms (Windows/Mac) run fine.
This doesn't occur when rearranging within the same window as shown in the first few seconds of the example.

**Comment by githubuser0xFFFF** [comment]
Thank you for this bug report. Ubuntu 22.04 and Wayland are not supported by ADS yet.

**Comment by nyoungbq** [comment]
Understood. Thank you for the quick response.

**Comment by githubuser0xFFFF** [comment]
I created an Ubuntu 22.04 VM with Qt6 to reproduce the problem. I can reproduce the problem. I think, this is causes by the use of Wayland by Ubuntu 22.04. Set the following ADS config flag, to switch to QWidget titlebars:

```c++
CDockManager::setConfigFlag(CDockManager::FloatingContainerForceQWidgetTitleBar, true);
```

**Comment by nyoungbq** [comment]
For my application we are currently utilizing QT 5.15 LTS, is Qt6 required for this native windows patch to work? [If you are unsure, no worries I can pull and test on my own]

**Comment by githubuser0xFFFF** [comment]
No, it also works with Qt 5.15. It does the same like setting the flag:

```c++
CDockManager::setConfigFlag(CDockManager::FloatingContainerForceQWidgetTitleBar, true);
```

**Comment by nyoungbq** [comment]
Alright, thanks so much!

**Comment by MoreTore** [comment]
I have this issue on wsl2 ubuntu 20.04.5
